lib/smbios.c

```diff
@@ -572,10 +621,36 @@ ulong write_smbios_table(ulong addr)
 
 	ctx.node = ofnode_null();
 	if (IS_ENABLED(CONFIG_OF_CONTROL)) {
```

Comment:
I think this should go away too. Because when you move DT stuff to specific driver this condition doesn't matter here.